hail/src/main/scala/is/hail/expr/ir/Sym.scala

```diff
@@ -0,0 +1,57 @@
+package is.hail.expr.ir
+
+import is.hail.utils.StringEscapeUtils
+
+// FIXME Sym instead of Symbol so we don't conflict with Scala's Symbol which is currently used by by the IR DSL
+object Sym {
+  var counter: Int = 0
+
+  def gen(base: String): Sym = {
+    counter += 1
+    Generated("j", base, counter)
+  }
+}
+
+abstract class Sym
+
+case class Identifier(name: String) extends Sym {
+  override def toString: String = {
+    if (name.matches("""\p{javaJavaIdentifierStart}\p{javaJavaIdentifierPart}*"""))
+      name
+    else
+      s"`${ StringEscapeUtils.escapeString(name, backticked = true) }`"
+  }
+}
+
+// lang is one of "py" or "j"
+case class Generated(lang: String, base: String, i: Int) extends Sym {
+  override def toString: String = s":$lang$base-$i"
```

Comment:
Should you assert `lang` is one of "py" or "j"?

Reply:
No, that's unnecessary.  For example, we could imagine generating symbols in C++ or R or whatever.  Each language should have a symbol generator, and they all need to have distinct values for `lang` so they don't actually collide.